### PR TITLE
Add clamp func

### PR DESCRIPTION
**Pull request by gusandrioli**
## Context
While substituting `GetLazyRootDirectory` funcs on `lazygit` I faced a conflict on `pkg/integration/clients/tui.go` as it's using `Clamp` from `lazygit/pkg/utils`. So adding this here and updating both would solve the issue and contribute to the goal of this package.

### CODE REVIEW

**Comment by gusandrioli** [comment]
@jesseduffield This is ready for review :)

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -22,6 +22,16 @@ func Max(x, y int) int {
 	return y
 }
 
+// Clamp returns a value x restricted between min and max
+func Clamp(x int, min int, max int) int {
```
Comment (jesseduffield) [inline pkg/utils/utils.go:26]: Just checking: is there a TestClamp in lazygit that we want to bring across?

Comment (gusandrioli) [reply pkg/utils/utils.go:26]: There isn't ([file](https://github.com/gusandrioli/lazygit/blob/master/pkg/utils/utils_test.go)). But i can add one quickly.

File: pkg/utils/utils_test.go [hunk 2]
```diff
@@ -33,7 +33,45 @@ func TestMin(t *testing.T) {
...
+			10,
+			10,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, tt.want, Clamp(tt.x, tt.min, tt.max))
+		})
 	}
```
Comment (gusandrioli) [inline pkg/utils/utils_test.go:75]: @jesseduffield Added a test now covering all cases!

File: pkg/utils/utils_test.go [hunk 3]
```diff
@@ -37,6 +37,44 @@ func TestMin(t *testing.T) {
...
+			10,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, tt.want, Clamp(tt.x, tt.min, tt.max))
+		})
+	}
+}
```
Comment (gusandrioli) [inline pkg/utils/utils_test.go:76]: @jesseduffield Added tests now!

**Comment by jesseduffield** [comment]
Nice one @gusandrioli

**Pull request merged**